src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1589,34 +1592,6 @@ private static string DeterminePropertyNameFieldName(string effectiveJsonPropert
                 return null;
             }
 
-            private static string GetTypeInfoPropertyName(ITypeSymbol type)
```

Comment:
Changing the identifier generation algorithm can be a potential source breaking change, since user code referencing source generated members from earlier SG versions could fail to compile.

Even if the new logic is equivalent to the old one, this will get backported to RC2 and it's not clear to me why we should take the risk this late.